### ISSUE TITLE
Low accuracy with soft-nms

### ISSUE THREAD

**Issue by tuyenqv** [tensorflow/models#8511]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [ ] I am using the latest TensorFlow Model Garden release and TensorFlow 2.
- [x ] I am reporting the issue to the correct repository. (Model Garden official or research directory)
- [x ] I checked to make sure that this issue has not already been filed.

## 1. The entire URL of the file you are using

https://github.com/tensorflow/models/tree/master/research/...

## 2. Describe the bug

I use soft-nms with iou_threshold=0.3 and sigma =0.5, but after training i observed that its performance is not as good as hard NMS

## 3. Steps to reproduce

I use sdd_resnet50_fpn with the following changes
```
    post_processing {
      batch_non_max_suppression {
        score_threshold: 1e-8
        iou_threshold: 0.3
        soft_nms_sigma: 0.5
        max_detections_per_class: 100
        max_total_detections: 100
      }
      score_converter: SIGMOID
    }
```
## 4. Expected behavior

According to the paper of soft-nms, its mAP should be higher than its hard nms counterpart

## 5. Additional context

N/A

## 6. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- Mobile device name if the issue happens on a mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 1.15.0
- Python version: 3.7.7
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: 10.1
- GPU model and memory: GTX 1080Ti
